packages/tizen_log/lib/tizen_log.dart

```diff
@@ -0,0 +1,160 @@
+      {String? file, String? func, int? line}) {
+    _log(_LogPriority.debug, tag, message, file: file, func: func, line: line);
+  }
+
+  /// Sends log with INFO priority and tag.
+  ///
+  /// [file], [func] and [line] parameters are set automatically if debug mode
+  /// is enabled, but they can be explicitly overridden.
+  static void info(String tag, String message,
+      {String? file, String? func, int? line}) {
+    _log(_LogPriority.info, tag, message, file: file, func: func, line: line);
+  }
+
+  /// Sends log with WARN priority and tag.
+  ///
+  /// [file], [func] and [line] parameters are set automatically if debug mode
+  /// is enabled, but they can be explicitly overridden.
+  static void warn(String tag, String message,
+      {String? file, String? func, int? line}) {
+    _log(_LogPriority.warn, tag, message, file: file, func: func, line: line);
+  }
+
+  /// Sends log with ERROR priority and tag.
+  ///
+  /// [file], [func] and [line] parameters are set automatically if debug mode
+  /// is enabled, but they can be explicitly overridden.
+  static void error(String tag, String message,
+      {String? file, String? func, int? line}) {
+    _log(_LogPriority.error, tag, message, file: file, func: func, line: line);
+  }
+
+  /// Sends log with FATAL priority and tag.
+  ///
+  /// [file], [func] and [line] parameters are set automatically if debug mode
+  /// is enabled, but they can be explicitly overridden.
+  static void fatal(String tag, String message,
+      {String? file, String? func, int? line}) {
+    _log(_LogPriority.fatal, tag, message, file: file, func: func, line: line);
+  }
+
+  static void _log(_LogPriority priority, String tag, String message,
+      {String? file, String? func, int? line}) {
+    if (isDebugEnabled) {
+      if (file == null || func == null || line == null) {
+        const int frameIndex = 3;
+        final _StackFrame frame = _stackTrace().firstWhere(
+            (_StackFrame e) => e.index == frameIndex,
+            orElse: () => _StackFrame(0, '', '', 0, 0));
+        if (frame.index == frameIndex) {
+          file ??= frame.file;
+          func ??= frame.function;
+          line ??= frame.line;
+        }
+      }
+      message = '${file ?? "-"}: ${func ?? "-"}(${line ?? "-"}) > ' + message;
+    }
+    final Pointer<Utf8> tagPtr = tag.toNativeUtf8();
+    final Pointer<Utf8> formatPtr = '%s'.toNativeUtf8();
+    final Pointer<Utf8> messagePtr = message.toNativeUtf8();
+    _dlogPrint(priority.value, tagPtr, formatPtr, messagePtr);
+    malloc.free(tagPtr);
+    malloc.free(formatPtr);
+    malloc.free(messagePtr);
+  }
+
+  static List<_StackFrame> _stackTrace() {
+    final List<_StackFrame> frames = <_StackFrame>[];
+    try {
+      final Iterable<RegExpMatch> matches =
+          _stackTraceRegExp.allMatches(StackTrace.current.toString());
+      for (final RegExpMatch m in matches) {
+        final List<String?> g = m.groups(<int>[1, 2, 3, 4, 5]);
```

Comment:
nit: Please avoid abbreviations.
```suggestion
      for (final RegExpMatch match in matches) {
        final List<String?> groups = m.groups(<int>[1, 2, 3, 4, 5]);
```